apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -575,64 +548,33 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     DatabaseQuery.call!(db_client, device_update_query)
   end
 
-  def fetch_device_introspection_minors(db_client, device_id) do
-    introspection_minor_statement = """
-    SELECT introspection_minor
-    FROM devices
-    WHERE device_id=:device_id
-    """
-
-    introspection_minor_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(introspection_minor_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.consistency(:quorum)
-
-    with {:ok, result} <- DatabaseQuery.call(db_client, introspection_minor_query),
-         [introspection_minor: introspection_minors] when is_list(introspection_minors) <-
-           DatabaseResult.head(result) do
-      {:ok, Enum.into(introspection_minors, %{})}
-    else
-      [introspection_minor: nil] ->
-        {:ok, %{}}
+  def fetch_device_introspection_minors(realm, device_id) do
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
 
-      %{acc: _, msg: error_message} ->
-        Logger.warning("Database error: #{error_message}.")
-        {:error, :database_error}
+    query =
+      Device
+      |> select([d], d.introspection_minor)
+      |> where(device_id: ^device_id)
+      |> put_query_prefix(keyspace_name)
 
-      {:error, reason} ->
-        Logger.warning("Failed with reason #{inspect(reason)}.")
-        {:error, :database_error}
+    with minors when is_map(minors) <- Repo.fetch_one(query, consistency: :quorum) do
```

Comment:
If `device.introspection_minor` can be `nil` in the database, this may return `{:ok, nil}`, instead of `{:ok, %{}}` as before.